src/test/regress/expected/pg15.out

```diff
@@ -388,15 +388,113 @@ ON (true)
+ ructions wake fluffily fluffily final gifts! furiou
+ s boost blithely fluffily idle ideas? fluffily even pin
+(5 rows)
+
+-- print the same items using a different regexp function
+SELECT o_comment FROM public.orders WHERE regexp_like(o_comment, 'fluffily.*fluffily') ORDER BY 1;
+                                 o_comment
+---------------------------------------------------------------------
+ al, bold deposits cajole fluffily fluffily final foxes. pending ideas beli
+ ly regular packages are fluffily even ideas. fluffily final
+ ng instructions integrate fluffily among the fluffily silent accounts. bli
+ ructions wake fluffily fluffily final gifts! furiou
+ s boost blithely fluffily idle ideas? fluffily even pin
+(5 rows)
+
+-- print the position where we find the second fluffily in the comment
+SELECT o_comment, regexp_instr(o_comment, 'fluffily.*(fluffily)') FROM public.orders ORDER BY 2 desc LIMIT 5;
+                                 o_comment                                  | regexp_instr
+---------------------------------------------------------------------
+ ng instructions integrate fluffily among the fluffily silent accounts. bli |           27
+ al, bold deposits cajole fluffily fluffily final foxes. pending ideas beli |           26
+ ly regular packages are fluffily even ideas. fluffily final                |           25
+ s boost blithely fluffily idle ideas? fluffily even pin                    |           18
+ ructions wake fluffily fluffily final gifts! furiou                        |           15
+(5 rows)
+
+-- print the substrings between two `fluffily`
+SELECT regexp_substr(o_comment, 'fluffily.*fluffily') FROM public.orders ORDER BY 1 LIMIT 5;
+         regexp_substr
+---------------------------------------------------------------------
+ fluffily among the fluffily
+ fluffily even ideas. fluffily
+ fluffily fluffily
+ fluffily fluffily
+ fluffily idle ideas? fluffily
+(5 rows)
+
+-- replace second `fluffily` with `silkily`
+SELECT regexp_replace(o_comment, 'fluffily', 'silkily', 1, 2) FROM public.orders WHERE regexp_like(o_comment, 'fluffily.*fluffily') ORDER BY 1 desc;
+                              regexp_replace
+---------------------------------------------------------------------
+ s boost blithely fluffily idle ideas? silkily even pin
+ ructions wake fluffily silkily final gifts! furiou
+ ng instructions integrate fluffily among the silkily silent accounts. bli
+ ly regular packages are fluffily even ideas. silkily final
+ al, bold deposits cajole fluffily silkily final foxes. pending ideas beli
+(5 rows)
+
+-- test numeric types with negative scale
+CREATE TABLE numeric_test(id numeric(3,-1), orig_value int);
+SELECT create_distributed_table('numeric_test','id');
+ create_distributed_table
+---------------------------------------------------------------------
+
+(1 row)
+
+INSERT into numeric_test SELECT x,x FROM generate_series(111, 115) x;
+SELECT * FROM numeric_test ORDER BY 1,2;
```

Comment:
I think we do have a critical issue with shard pruning on this:
```SQL



CREATE TABLE numeric_test(id numeric(3,-1), orig_value int);
SELECT create_distributed_table('numeric_test','id');
INSERT into numeric_test SELECT x,x FROM generate_series(111, 115) x;

SELECT * FROM numeric_test;
┌─────┬────────────┐
│ id  │ orig_value │
├─────┼────────────┤
│ 120 │        115 │
│ 110 │        112 │
│ 110 │        113 │
│ 110 │        114 │
│ 110 │        111 │
└─────┴────────────┘
(5 rows)

-- OPPPPS where is the row?
 SELECT * FROM numeric_test  WHERE id  = 120;
┌────┬────────────┐
│ id │ orig_value │
├────┼────────────┤
└────┴────────────┘
(0 rows)

SELECT undistribute_table('numeric_test');
NOTICE:  creating a new table for public.numeric_test
NOTICE:  moving the data of public.numeric_test
NOTICE:  dropping the old public.numeric_test
NOTICE:  renaming the new table to public.numeric_test
┌────────────────────┐
│ undistribute_table │
├────────────────────┤
│                    │
└────────────────────┘
(1 row)


 SELECT * FROM numeric_test  WHERE id  = 120;
┌─────┬────────────┐
│ id  │ orig_value │
├─────┼────────────┤
│ 120 │        115 │
└─────┴────────────┘
(1 row)

```

For example, if you do `INSERT into numeric_test SELECT x,x FROM generate_series(111, 115) x;` before create-dist-table, this works.

Needs more investigation.

Could be related to https://github.com/citusdata/citus/pull/2640 or needs something more for this type

Reply:
I tried to reproduce a similar issue with earlier versions of PG today. I was not able to do that. My guess is that we need negative scales to have this problem.

Reply:
If the fix is non-trivial, it is acceptable to throw an error if dist key is this non-supported numeric type. We can consider prioritizing that if/when needed

Reply:
Also, what about repartitioning on such columns? If that also doesn't work, I'm fine with not allowing create dist / ref table if the table include these columns